src/Microsoft.OData.Core/ODataBatchOperationRequestMessage.cs

```diff
@@ -57,7 +76,9 @@ public sealed class ODataBatchOperationRequestMessage : IODataRequestMessage, IO
             string contentId,
             IODataPayloadUriConverter payloadUriConverter,
             bool writing,
-            IServiceProvider container)
+            IServiceProvider container,
+            IList<string> dependsOnIds,

```

Comment:
>IList<string> [](start = 12, length = 13)

The input parameter should be an IEnumerable<string>. It doesn't need to be an IList, we copy it to our own list anyway (as we should).